build_defs/build_defs.bzl

```diff
@@ -39,17 +44,60 @@ def arcs_cc_schema(name, src, out = None):
         file_extension = ".h",
     )
 
-def arcs_kt_schema(name, src, out = None):
+def _wasm_name(name):
+    return name + "_wasm"
+
+def _wasm_deps(deps):
+    return [_wasm_name(dep) for dep in deps]
+
+def _jvm_and_wasm_library(name, srcs, deps):
+    kt_android_library(
+        name = name,
+        srcs = srcs,
+        deps = deps
+    )
+
+    kt_wasm_library(
+        name = _wasm_name(name),
+        srcs = srcs,
+        deps = _wasm_deps(deps)
+    )
+
+def arcs_kt_schema(name, src, out = None, deps = [], wasm_deps = []):
     """Generates a Kotlin file for the given .arcs schema file."""
+    out = out or _output_name(src, ".kt")
     _run_schema2pkg(
-        name = name,
+        name = name + "_genrule",
         src = src,
         out = out,
         language_name = "Kotlin",
         language_flag = "--kotlin",
         file_extension = ".kt",
     )
 
+    return out
+
+def arcs_manifest(name, srcs, deps = []):
```

Comment:
This seems to be named really generally, but doesn't apply to cpp-->wasm.

Reply:
It seems like you could just add the emscripten rules here without much fuss.

Reply:
I understand if you intend to do this in a later PR.

Reply:
Plan to do it in a later PR.

Reply:
I think this should have kt in the name. It looks like it's producing Kotlin code by default. When you add C++ in your follow-up PR, it'll start producing both Kotlin and C++, which seems unnecessary if you only actually want to use one of those two languages. `arcs_kt_manifest` and `arcs_cc_manifest` seem sensible to me.

Reply:
The way blaze/bazel works, unused rules are cheap and pruned from the graph. The original purpose of this macro was actually to be a rule which propagates manifests through a provider, ending in a dataflow check, with the compilation targets materializing via aspects. 

This is modeled on how new proto_library works. You build up a dependency graph via proto_library rules, which don't do any code-gen. And then you use a language-specific rule, XX_proto_library on the root target to do all the code gen. 

I was more or less going to hide that final step behind arcs_XX_particle, that is, if you have

arcs_kt_particle(name = 'foo', deps = [':foo-manifest'])

Then the deps attribute can be defined with an aspect() rule that takes the foo-manifest, and materializes the .kt files. And you can implement as many languages as you want, simply by adding more aspects. Alas, compiling code from aspects requires access to sandwich functions, and right now, sandwich functions only exist for Java and CPP, but not Kotlin, so I had to shelve this plan in favor of macros. 

But eventually I planned to remove this implementation, and for arcs_manifest() purely to be a transitive-dep gathering provider rule. The advantage of using aspects is that it not only removes explicitly DRY in declaring language, but it also potentially helps with de-duping transitive code-gen, since each manifest rule doesn't separately gen-code (which potentially includes other code)

This happens in protos where proto A includes proto B, and another rule exists which targets proto B. In old_proto_library, this would lead to many copies of proto B being compiled and included in the classpath.

Reply:
I've disabled the Android targets for now, since DevEx only needs wasm. I'll restore them later when Particle Accelerator work starts, hopefully by nuking the macros in favor of Starlark rules.